stdlib/regex.gr

```diff
@@ -4045,3 +4044,72 @@ export let replaceAll =
   ) => {
   regexReplaceHelp(rx, toSearch, replacement, true)
 }
+
+let regexSplitHelp = (rx: RegularExpression, str: String, all: Bool) => {
+  let buf = makeMatchBuffer(str)
+  let rec loop = searchPos => {
```

Comment:
I do wish that this logic was tail-recursive, but I don't think that needs to block this PR.

Reply:
#1474

Reply:
sorry about that I wanted to get it in so I basically just modified the replace helper and I figured it could be optimized later.